crates/aiken-lang/src/ast.rs

```diff
@@ -322,6 +322,7 @@ pub struct Validator<T, Expr> {
     pub doc: Option<String>,
     pub end_position: usize,
     pub fun: Function<T, Expr>,
+    pub other_fun: Option<Function<T, Expr>>,
```

Comment:
Would it make sense to consider having a list of functions as an internal representation?

Reply:
I thought of that as well, And the truth is there WILL always be at least ONE, so instead of have access patterns around `func_vec[0]` or `func_vec.get(0)` which can be awkward, we can directly use the field. We also know it will never be more than two. So the plain english description of a validator really is `at least one function with an _optional_ second function`. That being said if we did use a `Vec` and we wanted to have this notion of ALWAYS AT LEAST ONE, then we could use `Vec1` but so far I'm fairly have this the kind of access patterns this second field is yielding.

Reply:
Yes. It's definitely more type-safe this way. I just wondered if generalizing would have made sense. But it's probably just mental masturbation at this stage.